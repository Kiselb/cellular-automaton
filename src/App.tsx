import React, { FC } from "react";

// Never use BrowserRouter on GitHub Pages. There are some issues with it, it always shows blank screen. Use HashRouter instead, that will most probably work
import { Routes, Route, HashRouter as Router } from "react-router-dom";

import Main from "./pages/main/Main.page";
import SignIn from "./pages/signin/SignIn.page";

import { AuthProvider, RequireAuth } from "./services/auth/Auth";
import { StoreProvider } from "./services/store/StoreProvider";
import { AppStoreMode } from "./domain/types";

type Props = {
  probe?: ({}) => void;
};
const App: FC<Props> = ({ probe }) => {
  const [mode, setMode] = React.useState<AppStoreMode>("Native");
  const onModeChange = (storeMode: AppStoreMode) => {
    !!probe && probe({});
    setMode(storeMode as AppStoreMode);
  };

  return (
    <StoreProvider mode={mode}>
      <Router>
        <AuthProvider>
          <Routes>
            <Route
              path="/"
              element={
                <RequireAuth>
                  <Main mode={mode} onModeChange={onModeChange} />
                </RequireAuth>
              }
            />
            <Route path="/login" element={<SignIn />} />
          </Routes>
        </AuthProvider>
      </Router>
    </StoreProvider>
  );
};

export default App;
