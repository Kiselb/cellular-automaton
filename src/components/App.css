body {
    margin: 0;
    padding: 0;    
}
.app {
    width: 1000px;
    height: 100vh;
    margin: 0 auto;
    box-sizing: border-box;
    display: grid;
    grid-template-areas: "control panel";
    grid-template-rows: auto;
    grid-template-columns: 10em 1fr;
    grid-gap: 10px;
}
.control {
    grid-area: control;
}
.epoch {
    font-size: 1.25em;
}
.knob-label {
    color: lightgray;
    margin-top: 1.5em;
    margin-bottom: 0.5em;
}
.lifecontainer {
    grid-area: panel;
    overflow: scroll;
}
