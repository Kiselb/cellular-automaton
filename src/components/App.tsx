import React, { useEffect } from "react";
import { Provider } from "react-redux";

// Never use BrowserRouter on GitHub Pages. There are some issues with it, it always shows blank screen. Use HashRouter instead, that will most probably work
//
import { Routes, Route, HashRouter as Router } from "react-router-dom";

import { Main } from "./main/Main";
import SignIn from "./signin/SignIn";
import store from "./ducks/store";

import { AuthProvider, RequireAuth } from "./Auth";

const App = () => {
  const [user, setUser] = React.useState(
    () => localStorage.getItem("cellular-automaton.user") || ""
  );
  useEffect(() => {
    localStorage.setItem("cellular-automaton.user", user);
  }, [user]);

  return (
    <div className="container">
      <Provider store={store}>
        <Router>
          <AuthProvider>
            <Routes>
              <Route
                path="/"
                element={
                  <RequireAuth>
                    <Main />
                  </RequireAuth>
                }
              />
              <Route path="/login" element={<SignIn />} />
            </Routes>
          </AuthProvider>
        </Router>
      </Provider>
    </div>
  );
};

export default App;
