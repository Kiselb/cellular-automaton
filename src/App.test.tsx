import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

import App from "./App";

it("App renders correctly", () => {
  const { asFragment } = render(<App />);
  screen.debug();
  expect(asFragment()).toMatchSnapshot();
});
it("Set user", () => {
  const probe = jest.fn();
  localStorage.setItem("cellular-automaton.user", "Test");
  render(<App probe={probe} />);
  const control = screen.getByTestId("changeMode");
  fireEvent.change(control);
  expect(probe).toBeCalled();
});
