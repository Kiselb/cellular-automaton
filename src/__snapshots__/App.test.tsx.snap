// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <form
    class="username"
    data-testid="usernameform"
  >
    <input
      class="input"
      data-testid="username"
      placeholder="Введите Ваше имя"
      type="text"
    />
    <input
      class="input"
      data-testid="usernamesubmit"
      type="submit"
      value="Старт"
    />
  </form>
</DocumentFragment>
`;
